azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -811,141 +841,131 @@ def start_extension_command(self, extension_name, command, cmd_name, timeout, sh
+            This method ensure limits set with systemtctl at runtime
             TODO: set memory quotas
             """
             if self.enabled() and services_list is not None:
                 for service in services_list:
                     service_name = service.get('name', None)
                     unit_file_path = systemd.get_unit_file_install_path()
                     if service_name is not None and unit_file_path is not None:
-                        files_to_create = []
+                        # remove drop files from disk, new agent use systemdctl set-property
+                        files_to_remove = []
                         drop_in_path = os.path.join(unit_file_path, "{0}.d".format(service_name))
                         drop_in_file_cpu_accounting = os.path.join(drop_in_path,
                                                                    _DROP_IN_FILE_CPU_ACCOUNTING)
-                        files_to_create.append((drop_in_file_cpu_accounting, _DROP_IN_FILE_CPU_ACCOUNTING_CONTENTS))
+                        files_to_remove.append(drop_in_file_cpu_accounting)
                         drop_in_file_memory_accounting = os.path.join(drop_in_path,
                                                                       _DROP_IN_FILE_MEMORY_ACCOUNTING)
-                        files_to_create.append(
-                            (drop_in_file_memory_accounting, _DROP_IN_FILE_MEMORY_ACCOUNTING_CONTENTS))
-
-                        cpu_quota = service.get('cpuQuotaPercentage', None)
-                        if cpu_quota is not None:
-                            cpu_quota = str(cpu_quota) + "%"
-                            log_cgroup_info("Ensuring the {0}'s CPUQuota is {1}".format(service_name, cpu_quota))
-                            drop_in_file_cpu_quota = os.path.join(drop_in_path, _DROP_IN_FILE_CPU_QUOTA)
-                            cpu_quota_contents = _DROP_IN_FILE_CPU_QUOTA_CONTENTS_FORMAT.format(cpu_quota)
-                            files_to_create.append((drop_in_file_cpu_quota, cpu_quota_contents))
-
-                        self.__create_all_files(files_to_create)
-                        self.__reload_systemd_config()
-
-        def __reset_extension_services_cpu_quota(self, services_list):
+                        files_to_remove.append(drop_in_file_memory_accounting)
+
+                        drop_in_file_cpu_quota = os.path.join(drop_in_path, _DROP_IN_FILE_CPU_QUOTA)
+                        files_to_remove.append(drop_in_file_cpu_quota)
+                        self._cleanup_all_files(files_to_remove)
+
+                        cpu_quota = service.get('cpuQuotaPercentage')
+                        cpu_quota = "{0}%".format(cpu_quota) if cpu_quota is not None else ""  # setting an empty value resets to the default (infinity)
+                        properties_to_update = self._get_unit_properties_requiring_update(service_name, cpu_quota)
```

Comment:
why the extra check for the properties instead of just setting them up with no check?